### PR TITLE
feat(service-portal): Update disability license client

### PR DESCRIPTION
**Pull request by thorkellmani**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by alexdiljar — APPROVED** [review]
lgtm

File: libs/api/domains/disability-license/src/lib/disability-license.module.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
-import { DisabilityLicenseClientModule } from '@island.is/clients/disability-license'
+import {
+  DisabilityLicenseApiProvider,
+  DisabilityLicenseClientModule,
+} from '@island.is/clients/disability-license'
 import { DisabilityLicenseResolver } from './disability-license.resolver'
 
 @Module({
-  providers: [DisabilityLicenseResolver],
+  providers: [DisabilityLicenseResolver, DisabilityLicenseApiProvider],
```
Comment (eirikurn) [inline libs/api/domains/disability-license/src/lib/disability-license.module.ts:9]: Modules should not provide stuff from other modules. Either it's exported or not.

File: libs/clients/disability-license/src/lib/disabilityLicenseClient.provider.ts [hunk 2]
```diff
@@ -8,16 +8,17 @@ import {
 } from '@island.is/nest/config'
 import { DisabilityLicenseClientConfig } from './disabilityLicenseClient.config'
 import { createEnhancedFetch } from '@island.is/clients/middlewares'
+import { DisabilityLicenseService } from './disabilityLicenseClient.service'
 
-export const DisabilityLicenseApiProvider: Provider<DefaultApi> = {
-  provide: DefaultApi,
+export const DisabilityLicenseApiProvider: Provider<DisabilityLicenseService> = {
+  provide: DisabilityLicenseService,
```
Comment (eirikurn) [inline libs/clients/disability-license/src/lib/disabilityLicenseClient.provider.ts:14]: I'd suggest reverting changes to this file and let NestJS pass DefaultApi to the DisabilityLicenseService constructor.

You can stop exporting DefaultApi and DisabilityLicenseApiProvider from the client library. The client module can provide both DisabilityLicenseApiProvider and DisabilityLicenseService but only export DisabilityLicenseService.

So in other modules you import the DisabilityLicenseModule, and then get DisabilityLicenseService injected where its needed.

**Review by eirikurn — CHANGES_REQUESTED** [review]
Feel free to ping me on slack if something is unclear. 👍

File: libs/clients/disability-license/src/lib/disabilityLicenseClient.provider.ts (hunk 2 above)
Comment (thorkellmani) [reply libs/clients/disability-license/src/lib/disabilityLicenseClient.provider.ts:14]: yeah, that makes more sense

**Review by eirikurn — APPROVED** [review]

**Pull request merged**